roles/openshift_logging/files/generate-jks.sh

```diff
@@ -0,0 +1,21 @@
+#! /bin/sh
+
+set -ex
+
+keytool -genkey -alias {{component}} -keystore {{generated_certs_dir}}/{{component}}.jks -keypass kspass -storepass kspass \
+-keyalg RSA -keysize 2048 -validity 712 -dname "CN={{component}}, OU=OpenShift, O=Logging" {{extensions.stdout}}
+
+keytool -certreq -alias {{component}} -keystore {{generated_certs_dir}}/{{component}}.jks -storepass kspass \
+-file {{generated_certs_dir}}/{{component}}-jks.csr -keyalg RSA -dname "CN={{component}}, OU=OpenShift, O=Logging" {{extensions.stdout}}
+
+# this currently fails since we don't provide openssl as part of the ES image...
+openssl ca -in {{generated_certs_dir}}/{{component}}-jks.csr -notext -out {{generated_certs_dir}}/{{component}}-jks.crt \
+-config {{generated_certs_dir}}/signing.conf -extensions v3_req -batch -extensions server_ext
```

Comment:
unfortunately it looks like we're limited to what we can specify for creating using `oadm ca`. We need to be able to specify our dname so that Searchguard can verify roles and we need to be able to provide an oid. It looks like ca create-server-cert will make any `--hostname` entry a DNS entry...

```
oadm ca create-server-cert --signer-cert=$DIR/ca.crt --signer-key=$DIR/ca.key --signer-serial=$DIR/ca.serial.txt --key=$DIR/test.key --cert=$DIR/test.crt --hostnames='localhost, 127.0.0.1, oid:1.2.3.4.5.5'
```
```
Signature Algorithm: sha256WithRSAEncryption
        Issuer: CN=logging-signer-test
        Validity
            Not Before: Nov 21 22:48:19 2016 GMT
            Not After : Nov 21 22:48:20 2018 GMT
        Subject: CN= 127.0.0.1
        Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
                Public-Key: (2048 bit)
                Modulus:
                    00:a9:76:cb:5a:f5:77:d2:28:c4:ad:58:7e:85:fe:
                    dc:79:05:e1:b3:61:d9:2a:d4:9a:82:94:46:a1:d3:
                    3c:3d:51:78:20:cc:a7:ca:fe:43:30:19:38:3e:4a:
                    4e:2e:26:9d:5e:bc:7f:bf:c2:65:49:b1:5c:1e:6a:
                    4c:6a:4d:ab:da:fd:74:86:8a:2f:18:5b:41:ed:a9:
                    1a:cf:88:6b:37:50:47:d7:25:01:de:37:10:87:80:
                    12:44:72:65:0d:18:20:b4:ad:97:0e:fd:e4:79:38:
                    d1:23:35:c6:ad:6b:bc:1f:1c:52:7c:b0:1b:66:04:
                    ed:e7:92:79:77:ec:91:16:c7:a4:02:f7:ed:af:67:
                    b0:02:d1:84:a6:cb:a1:03:b8:56:7a:16:7f:a6:70:
                    43:34:bb:56:a8:90:d4:d9:af:2e:51:08:43:56:e3:
                    c7:08:73:38:39:60:a7:2e:81:d6:5a:76:8f:65:1a:
                    b6:ec:d7:73:9c:f1:22:87:0a:31:2e:fb:5c:49:d9:
                    5e:a5:74:41:e8:dd:c0:48:f1:8b:97:33:03:71:97:
                    df:1e:02:77:6a:2a:d3:19:17:6a:63:92:0d:86:ce:
                    c0:77:c2:7e:9d:7e:04:ee:4f:8a:1b:12:83:32:20:
                    0b:9e:9a:31:e2:f9:86:96:a0:1b:e1:cd:6a:6e:91:
                    df:b5
                Exponent: 65537 (0x10001)
        X509v3 extensions:
            X509v3 Key Usage: critical
                Digital Signature, Key Encipherment
            X509v3 Extended Key Usage: 
                TLS Web Server Authentication
            X509v3 Basic Constraints: critical
                CA:FALSE
            X509v3 Subject Alternative Name: 
                DNS: 127.0.0.1, DNS: oid:1.2.3.4.5.5, DNS:localhost
```